src/util/Synchronized.h

```diff
@@ -32,6 +33,19 @@ struct AllowsSharedLocking<
                    decltype(std::declval<M&>().unlock_shared())>>
     : std::true_type {};
 
+/// A very simple spin lock, stolen from cppreference
```

Comment:
/// A very simple spin lock, stolen from cppreference. A spin lock is simply a lock that waits (as long as is necessary) for the lock to be released before locking it. In particular, this is OK when using it for serializing concurrent accesses to an object that are very simple and fast.